src/ports/postgres/modules/deep_learning/madlib_keras.py_in

```diff
@@ -520,21 +544,35 @@ def fit_transition(state, dependent_var, independent_var, dependent_var_shape,
         and only gets cleared in eval transition at the last row of the last iteration.
 
     """
-    if not independent_var or not dependent_var:
+    if not dependent_var_shape:
```

Comment:
As I understand it, we require the input table to be minibatched--so every input row must contain an array with the same dimensions.  And where the sizes of all but the first dimension of the arrays must match between all rows.  A NULL value for a class should be okay, and maybe even a NULL image somewhere in the array (although I'm not sure what that would mean), but I think if either the x or y array itself were NULL, or the shape is NULL, that's an invalid input since it wasn't minibatched properly.

I'm not sure if we explicitly error out in the validation code during initialization in that case.  I figured if we were allowing that before is was probably unintentional.  Is there a reason why we would need to support that?

Reply:
Re: checking dependent_var vs dependent_var_shape, I don't have a strong preference.  I think I went with that out of a habit for avoiding messing with any potentially large data structures in favor of small ones, if all we need is a boolean.  But I don't think it actually matters in this case--if you like dependent_var better I can change it.

Reply:
Maybe the safest would be to check all 4 of them?  If any one of them is NULL but another of the 3 is non-NULL then that's a sign that there's something seriously wrong and we should probably just stop and error out immediately.

Reply:
you are right that x or y being null probably should be a case for exception since our minibatch code should have minibatched everything. 
For dependent_var vs dependent_var_shape, I'll leave it up to you to choose which one to check for (I'm leaning towards dependent_var)